### PR TITLE
Transpose following Enharmonic Rules & Normalize Chord Suffix'

### PR DESCRIPTION
**Pull request by isaiahdahl**
Open PR for this issue https://github.com/PraiseCharts/ChordChartJS/issues/45

- [x] add a failing test with all the numbers -> chord string cases
- [x] stop the normalizeSuffix from normalizing the Major/Minor suffix
- [x] add ability for Key to know whether it's major or Minor, different transpose distance whether it's major or minor
- [x] change how Key :: normalize() and useModifier() work to facilitate enharmonic rules 

normalize suffix
- [x] write tests for normalizing
- [x] fix other tests to not conflict with new normalizing (replace sus4 with sus)  
- [x] build scripts
- [x] review my approach with Martijn

### CODE REVIEW

**Comment by isaiahdahl** [comment]
Got lots that can be cleaned up here and we may want to take a different approach with the Key class for minor keys but this puts my thoughts on paper so to speak.

File: src/chord.js [hunk 1]
```diff
@@ -217,9 +204,9 @@ class Chord {
       bass = null,
     },
   ) {
-    this.root = root || new Key({ note: base, modifier });
     this.suffix = presence(suffix);
-    this.bass = bass || (bassBase ? new Key({ note: bassBase, modifier: bassModifier }) : null);
+    this.root = root || new Key({ note: base, modifier: modifier, minor: suffix === 'm' });
```
Comment (isaiahdahl) [inline src/chord.js:208]: Just seeing where this comes up. An option to start a discussion from my IDE popped up, gonna see what happens

**Comment by isaiahdahl** [comment]
Awesome! Looks good to me!

**Comment by martijnversluis** [comment]
@isaiahdahl I was just wondering about the task `change how Key :: normalize() and useModifier() work to facilitate enharmonic rules`. I guess when discussing it we concluded we'd merge `normalize` and `normalizeEnharmonics`? I think `useModifier` should not be impacted. What do you think?

**Comment by isaiahdahl** [comment]
@martijnversluis I think that's correct. We just have to make sure the order of operations is correct with normalize() and useModifier() 

My concern when I wrote the initial task was that a chord would be normalized, and then useModifier would un-normalize it.

File: src/key.js [hunk 2]
```diff
@@ -92,11 +93,11 @@ class Key {
 
     let numericKey = new Key({ note: 1 });
     let symbolKey = key.clone();
-    const reference = this.clone().normalize().useModifier(key.modifier);
+    const reference = this.clone().normalize().useModifier(key.modifier).normalizeEnharmonics(key);
```
Comment (isaiahdahl) [inline src/key.js:113]: Like here for example. If useModifier() was to be the last called function, it wouldn't be normalized right. So if we combine normaize() and normalizeEnharmonics(), we gotta make sure useModifier() isn't the last thing called.

**Comment by martijnversluis** [comment]
@isaiahdahl Do you think this PR is complete now? Is the `change how Key :: normalize() and useModifier() work to facilitate enharmonic rules` part covered?

**Comment by isaiahdahl** [comment]
I think that there is still at least some refactoring we should consider, but it could be in a later issue/task because everything is working as intended now and passing all the tests to satisfy the objective.

I think .normalize() and .normalizeEnharmonics(key) share some common functionality in the sense that they both facilitate the purpose of normalizing the chord.

An example of this is in the chord.js file
CleanShot 2022-02-17 at 09 46 05

On Line 245 `root: this.root.normalize().normalizeEnharmonics(key)` The first normalize is redundant. If I remove the `.normalise()` and just use the `.normalizeEnharmonics(key)` all the tests still pass.

I think we could merge so we can move forward on other tasks and then possibly open up a new issue that just focuses on a refactor / clean up of the normalizing. We had also briefly discussed putting some normalizing features in a config that could be turned on or off so maybe that would be a good time to tackle that as well.

Thoughts?

**Comment by martijnversluis** [comment]
Yeah makes sense! I'll remove the redundant `normalize()` and check if there are other quick wins. Other than that I'd like to merge it and move on, because this has been open for too long now. 👍

**Pull request merged**